walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,140 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    // The total (common) free time we have to conduct the meeting.
+    ArrayList<TimeRange> free_time = new ArrayList<TimeRange>();
```

Comment:
For variable names in Java it's common to use lowerCamelCase: `freeTime`, `meetingDuration`, ... Please double-check in your style guide (it could differ from the internal Java style guide.

For your repository, this is not as important to change as your code is self-consistent. When working on the team project next week, this will become an issue.

Reply:
Hmm okay, I'll make a note of it. I'm not sure but I saw somewhere saying that methods are in camel case but variables are used with dashes. So I started doing it this way.

I'll make sure to check it in the style guide next time.
Thanks for letting me know.

Reply:
I might be wrong about the style guide you're using. So please double-check.

Reply:
Ok I will

Reply:
"In general, use functionNamesLikeThis, variableNamesLikeThis, ClassNamesLikeThis, EnumNamesLikeThis, methodNamesLikeThis, and SYMBOLIC_CONSTANTS_LIKE_THIS."

Reply:
Okay I think I misinterpreted some text and messed up the whole naming thing in this class :x